### PR TITLE
Doc for in app prompts

### PR DESCRIPTION
**Pull request by kul3r4**
#### WHAT
Adds docs for in app prompts. Code is still evolving so will need a second pass soon after new features are merged.

#### WHY
It's missing documentation for the in app prompts

#### HOW
Creates some new doc pages

#### Checklist :clipboard:
- [ ] Add explicit visibility modifier and explicit return types for public declarations
- [ ] Run spotless check
- [ ] Run tests
- [ ] Update metalava's signature text files

### CODE REVIEW

File: docs/datalayer-phone-ui.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+The `shouldDisplayPrompt` method allows to check if the prompt should be displayed, based on the following conditions:
+
+- there is a watch connected
+
+- the app is not installed
+
+The `shouldDisplayPrompt` method is relying on the capability defined in the wear.xml of the Wear app, if the Wear app is installed 
+but hasn't been updated with the capability definition, this method may still return a `AppHelperNodeStatus`
+indicating that the Wear app is not installed.
+
```
Comment (luizgrp) [inline docs/datalayer-phone-ui.md:18]: should also add the [performAction](https://github.com/google/horologist/pull/2076) function

**Review by luizgrp — APPROVED** [review]
Thanks for improving these!
Left comments but happy to merge and get them addressed in future PRs.

File: datalayer/watch/src/main/java/com/google/android/horologist/datalayer/watch/WearDataLayerAppHelper.kt [hunk 2]
```diff
@@ -189,7 +189,8 @@ public class WearDataLayerAppHelper(
 
     /**
      * Marks that the necessary setup steps have been completed in the app such that it is ready for
-     * use. Typically this should be called when any pairing/login has been completed.
+     * use. Typically this should be called when any pairing/login has been completed. If used for
```
[the remark this answers is not in the document]
Comment (kul3r4) [reply datalayer/watch/src/main/java/com/google/android/horologist/datalayer/watch/WearDataLayerAppHelper.kt:192]: Associate the watch app with the phone app I think, didn't change this as it was already there but happy to just keep login

File: docs/datalayer-sample.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
+# DataLayer sample app
+
+The goal of this sample is to show how to use the DataLayer helpers both on phone and watch.
+In order to be able to demo all the features, you need to install the sample both on the watch and
```
[the remark this answers is not in the document]
Comment (kul3r4) [reply docs/datalayer-sample.md:4]: will remove

File: docs/datalayer-phone-ui.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+## ReEngage prompt
+Use the `ReEngagePrompt` to build the UI of a prompt to ask the user to install the app on their watch.
+
+## SignIn prompt
+Use the `SignInPrompt` to build the UI of a prompt to ask the user to finish the sign in on the watch.
```
[the remark this answers is not in the document]
Comment (kul3r4) [reply docs/datalayer-phone-ui.md:25]: added

File: docs/datalayer-phone-ui.md (hunk 1 above)
Comment (kul3r4) [reply docs/datalayer-phone-ui.md:18]: done

**Pull request merged**